identity_jose/src/jws/encoder.rs

```diff
@@ -0,0 +1,300 @@
+// Copyright 2020-2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use core::str;
+use serde::Serialize;
+use std::future::Future;
+
+use crate::error::Error;
+use crate::error::Result;
+use crate::jws::CharSet;
+use crate::jws::JwsFormat;
+use crate::jws::JwsHeader;
+use crate::jws::Recipient;
+use crate::jwu;
+
+/// The protected JWS header.
+pub type EncoderProtectedHeader = JwsHeader;
+/// The unprotected JWS header.
+pub type EncoderUnprotectedHeader = JwsHeader;
+/// The message to sign as a byte vector.
+pub type EncoderMessage = Vec<u8>;
```

Comment:
Is `Message`  standard JOSE parlance? If not it might be an idea to write a sentence about what to expect here.

Reply:
> JSON Web Signature (JWS)
>    A data structure representing a digitally signed or MACed message.
> https://www.rfc-editor.org/rfc/rfc7515#section-2

I think it's common terminology.